presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -151,7 +155,7 @@ public void addInput(Page page)
             aggregate.processPage(page);
             memorySize += aggregate.getEstimatedSize();
         }
-        if (partial) {
+        if (partial || useSystemMemoryPool) {
```

Comment:
The `partial` field is only used to do system allocations. Now that you introduce `useSystemMemoryPool`, is `partial` still needed?

Reply:
I changed it to only depend on the `useSystemMemoryPool` flag. I set `useSystemMemoryPool` in the `LocalExecutionPlanner` based on the `Step`